### DISCUSSION TITLE
Sending ITestOutputHelper messages to the console immediately

### DISCUSSION THREAD

**Discussion by Piedone** [xunit/xunit#2925] [Question] [answered]
We're investigating why some of our UI tests randomly hang under Windows GitHub Actions runners (see https://github.com/Lombiq/UI-Testing-Toolbox/issues/228). To help with this, we'd need to see output from the tests immediately as they are written. So, we need messages written with `ITestOutputHelper.WriteLine()` to be sent to the console immediately. Currently, this output is only sent when `dotnet test` completes (run as `dotnet test --logger 'console;verbosity=detailed'`).

Is this possible? I looked but couldn't find a solution. I'm not sure if this is something concerning xUnit or `dotnet test`.

**Comment by bradwilson** [comment, marked as answer]
There is an open issue for this: https://github.com/xunit/visualstudio.xunit/issues/408

I'll make sure it gets into a pre-release build soon.

**Reply by Piedone** [reply]
I see, thanks!

**Reply by bradwilson** [reply]
See the linked issue as this feature is now available in pre-release, and will be part of the final 2.8.1 release.
